src/Application.php

```diff
@@ -426,7 +446,10 @@ public function routeMiddleware(ServerRequestInterface $request, ResponseInterfa
             if ($result->isMethodFailure()) {
                 $response = $response->withStatus(405)
                     ->withHeader('Allow', implode(',', $result->getAllowedMethods()));
-                return $next($request, $response, 405);
+
+                return $this->raiseThrowables
+                    ? $response
+                    : $next($request, $response, 405);
```

Comment:
Should we use constant also here? (`StatusCodeInterface::STATUS_METHOD_NOT_ALLOWED`)